examples/react-app/src/App.tsx

```diff
@@ -14,19 +15,18 @@ function App() {
   const [tags, _setTags] = useState<string[]>([]);
   const [limit, _setLimit] = useState<number>(10);
 
-  const { data, error, refetch } = useDefaultServiceFindPets({ tags, limit });
+  const { data, error, refetch } = useFindPets({ tags, limit });
   // This is an example of using a hook that has all parameters optional;
   // Here we do not have to pass in an object
-  const { data: _ } = useDefaultServiceFindPets();
+  const { data: _ } = useFindPets();
```

Comment:
Dropping the use of tag names makes this a breaking change.

I must consider whether dropping the tag to generate the hook is helpful.
I see a benefit in having the tag help describe the hook, especially for large spec files.
I also see the beauty in not having the tag there.

Reply:
are you referring the "defaultService"? because wasn't sure what to put there if there's no real concept of a service in the new client.

Reply:
I was referring to defaultService. For large spec files it’s a nice to have buckets to put similar api calls. dogService, catService, etc.  makes DX better has you can discover api calls related to what you are working with.

How are the names of the hooks being generated now? I believe they are generated from the method names of the generated client code from @hey-api. Does @hey-api use operationId or is it generated from the displayName, or something else? How are duplicate names handled.

I can imagine 2 generated hooks having the same or similar name using the clients code now that tags are not used in the names.


That was my stream of consciousness. I haven’t dug into the code yet.

Reply:
@seriouslag it is using the operationId and AFAIK it should be unique in the spec no?

Reply:
I believe OperationId to be unique but not a required field in the OpenAPI spec.

Reply:
yeah I checked and it is either using the operationId, or build it using the URL, so we can definitely have a conflicts no?